### PR TITLE
v2.10.4: add some missing simple earn endpoints

### PR DESCRIPTION
**Pull request by SebastianBoehler**
## Summary
Added some simple earn endpoints as I saw https://github.com/tiagosiebler/binance/issues/386
I am confused on how to submit params for get request, proceeded with query params but there is some inconsistency across methods. sometimes as body, sometimes as url query. According to docs for get request it must be query string 

## Additional Information
Anyone feel free to contribute more, I've just added about half of the simple earn endpoints, those are important to me.

Tried to not include breaking changes and kept naming of existing methods, tho types were outdated, not accurate on some of them

### CODE REVIEW

File: src/main-client.ts [hunk 1]
```diff
@@ -1145,6 +1158,77 @@ export class MainClient extends BaseRestClient {
...
+   * Simple earn Endpoints
+   *
+   **/
+
+  getFlexibleSavingProducts(
+    params: SimpleEarnProductListParams,
+  ): Promise<SimpleEarnFlexibleProductListResponse> {
+    return this.getPrivate(
+      `/sapi/v1/simple-earn/flexible/list?${serialiseParams(params)}`,
+    );
```
Comment (tiagosiebler) [inline src/main-client.ts:1172]:
```suggestion
    return this.getPrivate(`/sapi/v1/simple-earn/flexible/list`, params);
```

**Review by tiagosiebler — CHANGES_REQUESTED** [review]
Thanks for the PR! For GET requests, simply pass any params as an object (in the second param of the getPrivate(url, params) call) - the networking abstraction will automatically handle serialisation etc while building the request. Here's an example from the master branch:
https://github.com/tiagosiebler/binance/blob/master/src/main-client.ts#L259-L261

File: src/main-client.ts (hunk 1 above)
Comment (SebastianBoehler) [reply src/main-client.ts]: Alright, updated :)

**Comment by SebastianBoehler** [comment]
By merging/approving https://github.com/tiagosiebler/binance/pull/420 first in case my commit was the only thing preventing it from merging we could have both changes in the version bump from this PR

File: src/types/simpleEarn.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
+export interface SimpleEarnProductListParams {
+  asset?: string;
+  current?: number;
+  size?: number;
+  recvWindow?: number;
+  timestamp: number;
```
Comment (tiagosiebler) [inline src/types/simpleEarn.ts:6]: Just saw these. So for private API calls, you don't need to worry about providing recvWindow nor timestamp in any of the request params. When the SDK builds the request and prepares the authentication part of it, it will:
- Make a timestamp (Date.now())
- Apply the recv window in the general REST client config
- Build a request signature using api credentials + timestamp
- Build a request using this signature, api key, timestamp (important to use the same timestamp!) and send it to the API.

Having this as a request param might cause interference or fail auth/sign completely (please test this after implementing these changes). So recvWindow & timestamp should be removed from any request params - they're already handled automatically by the SDK.

File: src/main-client.ts [hunk 3]
```diff
@@ -1145,6 +1158,67 @@ export class MainClient extends BaseRestClient {
...
   }
 
+  /**
+   *
+   * Simple earn Endpoints
+   *
+   **/
+
+  getFlexibleSavingProducts(
+    params: SimpleEarnProductListParams,
```
Comment (tiagosiebler) [inline src/main-client.ts:1168]: If all properties on this object are optional, then "params" should be optional too. Same for any other calls with fully optional params.
```suggestion
    params?: SimpleEarnProductListParams,
```

The SDK will automatically see that this is undefined and will make the request without any params (and it looks cleaner in code).

File: src/types/simpleEarn.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  asset?: string;
+  productId?: string;
+  current?: number;
+  size?: number;
+}
+
+export interface SimpleEarnLockedProductPositionParams
+  extends SimpleEarnFlexibleProductPositionParams {
+  positionId?: string;
+}
```
Comment (tiagosiebler) [inline src/types/simpleEarn.ts:89]: Since it's only 4 extra params (not 10-50), I would prefer a flatter view, even if it means a little more repetition:
```suggestion
export interface SimpleEarnLockedProductPositionParams {
  asset?: string;
  productId?: string;
  current?: number;
  size?: number;
  positionId?: string;
}
```

What might start with one simple `extends` can quickly become a mess that's hard to follow around (for some).

File: src/main-client.ts [hunk 5]
```diff
@@ -1145,6 +1158,67 @@ export class MainClient extends BaseRestClient {
...
+  ): Promise<SimpleEarnLockedProductPositionResponse> {
+    return this.getPrivate(`/sapi/v1/simple-earn/locked/position`, params);
+  }
+
+  getSimpleEarnAccount(params: {
+    recvWindow?: number;
+    timestamp: number;
+  }): Promise<SimpleEarnAccountResponse> {
+    return this.getPrivate(`/sapi/v1/simple-earn/account`, params);
+  }
```
Comment (tiagosiebler) [inline src/main-client.ts:1220]:
```suggestion
  getSimpleEarnAccount(): Promise<SimpleEarnAccountResponse> {
    return this.getPrivate(`/sapi/v1/simple-earn/account`);
  }
```

**Review by tiagosiebler — CHANGES_REQUESTED** [review]
Some changes. Pls double check which params should be optional

**Comment by SebastianBoehler** [comment]
All done :)

**Review by tiagosiebler — APPROVED** [review]
Looks great! Thanks for the updates!

**Comment by tiagosiebler** [comment]
Released :)
